tests/test.cpp

```diff
@@ -2841,6 +2843,40 @@ void FixedLengthArrayTest() {
 #endif
 }
 
+void NativeTypeTest() {
+// looks like VS10 does not support std::vector with explicit alignment
+// From stackoverflow (https://stackoverflow.com/questions/8456236/how-is-a-vectors-data-aligned):
+//
+// Visual C++ version 2010 will not work with an std::vector with classes whose alignment are specified.
+// The reason is std::vector::resize.
+// When compiling, next error appears:
+// c:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\include\vector(870): error C2719: 
```

Comment:
Where does this alignment requirement come from? You're telling me VS2010 can't do vectors with doubles in them? I somehow doubt that.

Either way, since this example is about native types and not about alignment, how about changing the example to use floats instead?

Reply:
No, the reason is the next:
flatc inserts manual alignment to structs:
`FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(8) Vector3D FLATBUFFERS_FINAL_CLASS {`
For double coordinates alignment is 8, for float it is 4.

For table `table ApplicationData {
  vectors:[Vector3D];
}`  object api generates `std::vector<Vector3D>` field.

Thus we have std::vector of manually alignment structs. MSVC before version 2012 does not compile it with error `c:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\include\vector(870): error C2719:  '_Val': formal parameter with __declspec(align('8')) won't be aligned [C:\projects\flatbuffers\flattests.vcxproj].` 
Internet says that it is because `std::vector::resize` method, which does not work with any alignments except 1. The issue was solved in MSVC 2012 and later. Can't rapidly find ticket for this issue  - some links are dead. If it is important, I will continue the search.

Reply:
I wouldn't mind resolving this, as we have several vectors of structs which don't have this problem, so I'd like to solve it at the core rather than using this #ifdef, since otherwise users will run into this.

What is the full error, i.e. where does the error above in the FlatBuffers code originate from? My guess would be in the implementation of `CreateVectorOfNativeStructs` where `std::transform` calls `resize`?

If so, can we replace that code with code that just does manual `emplace_back`, and only calls `resize` before it if not on VS2010?

Reply:
I have tried to compile just this minimal code with MSVC 2010:
`#include "stdafx.h"
#include <vector>

__pragma(pack(1)) 
struct __declspec(align(8)) Vector3D  
{
 public:
  double x_;
  double y_;
  double z_;
};

int _tmain(int argc, _TCHAR* argv[])
{
  std::vector<Vector3D> vectors;
  return 0;
}`

And I've got the same error:

`1>c:\program files (x86)\microsoft visual studio 10.0\vc\include\vector(870): error C2719: '_Val': formal parameter with __declspec(align('8')) won't be aligned
1>          c:\users\ageraskin.000\documents\visual studio 2010\projects\test\test\test.cpp(21) : see reference to class template instantiation 'std::vector<_Ty>' being compiled
1>          with
1>          [
1>              _Ty=Vector3D
1>          ]`

However, it works with align <= 4:

`
__pragma(pack(1)) 
struct __declspec(align(2)) Test
{
 public:
  int16_t a_;
  int8_t b_;
  int8_t padding0__;
};

int _tmain(int argc, _TCHAR* argv[])
{
  std::vector<Test> vectors;
  return 0;
}
`

Thus, Vector3D with float data types should work (align == 4), Will try now. 

However, compatibility with MSVC 2010 is not full actually.

Reply:
Your Native::Vector3D doesn't have any alignment spec though, so I still don't understand what is the core cause of this. If this is to be fixed, I'd like to know where it comes from.

But ok, if with floats we at least don't need all the `#ifdef`s, then that is fine for now.